tests/integration/olp-cpp-sdk-dataservice-read/VersionedLayerClientTest.cpp

```diff
@@ -2501,4 +2519,206 @@ TEST_F(DataserviceReadVersionedLayerClientTest, GetDataOnlineOnly) {
+  ASSERT_LT(0, data_response.GetResult()->size());
+  std::string data_string(data_response.GetResult()->begin(),
+                          data_response.GetResult()->end());
+  ASSERT_EQ("someData", data_string);
+
+  {
+    SCOPED_TRACE("Check if data was cached with previous request.");
+    auto future = client->GetData(request);
+    data_response = future.GetFuture().get();
+    ASSERT_TRUE(data_response.IsSuccessful());
+    SCOPED_TRACE("Next call should fail despite cached response.");
+    EXPECT_CALL(*network_mock_,
+                Send(IsGetRequest(kHttpLookupQuery), _, _, _, _))
+        .WillOnce(
+            ReturnHttpResponse(olp::http::NetworkResponse().WithStatus(429),
+                               "Server busy at the moment."));
+    future = client->GetData(request.WithFetchOption(OnlineOnly));
+    data_response = future.GetFuture().get();
+    ASSERT_FALSE(data_response.IsSuccessful());
+  }
+}
+
+TEST_F(DataserviceReadVersionedLayerClientTest, GetTileTwoSequentialCalls) {
+  olp::client::HRN hrn(GetTestCatalog());
+  auto client = std::make_unique<olp::dataservice::read::VersionedLayerClient>(
+      hrn, "testlayer", 4, *settings_);
+  {
+    EXPECT_CALL(*network_mock_,
+                Send(IsGetRequest(kHttpLookupQuery), _, _, _, _))
+        .WillOnce(ReturnHttpResponse(olp::http::NetworkResponse().WithStatus(
+                                         olp::http::HttpStatusCode::OK),
+                                     kHttpResponseLookupApiQuery));
+
+    EXPECT_CALL(*network_mock_,
+                Send(IsGetRequest(kHttpQueryTreeIndex), _, _, _, _))
+        .WillOnce(ReturnHttpResponse(olp::http::NetworkResponse().WithStatus(
+                                         olp::http::HttpStatusCode::OK),
+                                     kHttpSubQuads));
+
+    EXPECT_CALL(*network_mock_, Send(IsGetRequest(URL_LOOKUP_BLOB), _, _, _, _))
+        .WillOnce(ReturnHttpResponse(olp::http::NetworkResponse().WithStatus(
+                                         olp::http::HttpStatusCode::OK),
+                                     kHttpResponseLookupBlob));
+
+    EXPECT_CALL(*network_mock_,
+                Send(IsGetRequest(kHttpResponseBlobData_5904591), _, _, _, _))
+        .WillOnce(ReturnHttpResponse(olp::http::NetworkResponse().WithStatus(
+                                         olp::http::HttpStatusCode::OK),
+                                     "someData"));
+
+    auto request = olp::dataservice::read::TileRequest().WithTileKey(
+        olp::geo::TileKey::FromHereTile("5904591"));
+    auto data_response = client->GetData(request).GetFuture().get();
+
+    ASSERT_TRUE(data_response.IsSuccessful())
+        << ApiErrorToString(data_response.GetError());
+    ASSERT_LT(0, data_response.GetResult()->size());
+    std::string data_string(data_response.GetResult()->begin(),
+                            data_response.GetResult()->end());
+    ASSERT_EQ("someData", data_string);
+  }
+  SCOPED_TRACE(
```

Comment:
SCOPED_TRACE is not a comment. Is will be printed to the command line if this test fails. I dont think you want such a long string to be printed. Therefore use short sentences for the SCOPED_TRACES()

```cpp
SCOPED_TRACE("Neighboring tile, same subquad, check metadata cached");
```

Also as the name suggests the SCOPED_TRACE needs to be under the below scope.
```cpp
{
  SCOPED_TRACE("Scope trace 1");
}
{
  SCOPED_TRACE("Scope trace 2");
}
{
  SCOPED_TRACE("Scope trace 3");
}
```